src/ingestendpoint/secret.go

```diff
@@ -159,21 +162,23 @@ func (g *EndpointSecretGenerator) PrepareFields(ctx context.Context, dk *dynatra
 		return nil, errors.WithMessage(err, "failed to query tokens")
 	}
 
-	if token, ok := tokens.Data[dtclient.DynatraceDataIngestToken]; ok {
-		fields[TokenSecretField] = string(token)
-	}
+	if !dk.FeatureDisableMetadataEnrichment() {
+		if token, ok := tokens.Data[dtclient.DynatraceDataIngestToken]; ok {
+			fields[MetricsTokenSecretField] = string(token)
+		}
 
-	if diUrl, err := dataIngestUrl(dk); err != nil {
-		return nil, err
-	} else {
-		fields[UrlSecretField] = diUrl
+		if diUrl, err := dataIngestUrl(dk); err != nil {
```

Comment:
Done, but replaced suggestion of `parseDataIngestUrl` to `dataIngestUrlFor(dynakube)` because it's not parsing but rather calculating URL based on dynakube content.